exchanges/kucoin/kucoin.go

```diff
@@ -0,0 +1,2047 @@
+func (ku *Kucoin) GetCurrency(ctx context.Context, currency, chain string) (*CurrencyDetail, error) {
+	resp := struct {
+		Data CurrencyDetail `json:"data"`
+		Error
+	}{}
+	if currency == "" {
+		return nil, errors.New("currency can't be empty")
+	}
+	params := url.Values{}
+	if chain != "" {
+		params.Set("chain", chain)
+	}
+	return &resp.Data, ku.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues(kucoinGetCurrency+strings.ToUpper(currency), params), &resp)
+}
+
+// GetFiatPrice gets fiat prices of currencies, default base currency is USD
+func (ku *Kucoin) GetFiatPrice(ctx context.Context, base, currencies string) (map[string]string, error) {
+	resp := struct {
+		Data map[string]string `json:"data"`
+		Error
+	}{}
+	params := url.Values{}
+	if base != "" {
+		params.Set("base", base)
+	}
+	if currencies != "" {
+		params.Set("currencies", currencies)
+	}
+	return resp.Data, ku.SendHTTPRequest(ctx, exchange.RestSpot, common.EncodeURLValues(kucoinGetFiatPrice, params), &resp)
+}
+
+// GetMarkPrice gets index price of the specified pair
+func (ku *Kucoin) GetMarkPrice(ctx context.Context, pair string) (*MarkPrice, error) {
+	resp := struct {
+		Data MarkPrice `json:"data"`
+		Error
+	}{}
+	if pair == "" {
+		return nil, errors.New("pair can't be empty")
+	}
+	return &resp.Data, ku.SendHTTPRequest(ctx, exchange.RestSpot, fmt.Sprintf(kucoinGetMarkPrice, pair), &resp)
+}
+
+// GetMarginConfiguration gets configure info of the margin
+func (ku *Kucoin) GetMarginConfiguration(ctx context.Context) (*MarginConfiguration, error) {
+	resp := struct {
+		Data MarginConfiguration `json:"data"`
+		Error
+	}{}
+	return &resp.Data, ku.SendHTTPRequest(ctx, exchange.RestSpot, kucoinGetMarginConfiguration, &resp)
+}
+
+// GetMarginAccount gets configure info of the margin
+func (ku *Kucoin) GetMarginAccount(ctx context.Context) (*MarginAccounts, error) {
+	resp := struct {
+		Data MarginAccounts `json:"data"`
+		Error
+	}{}
+	return &resp.Data, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, kucoinGetMarginAccount, nil, &resp)
+}
+
+// GetMarginRiskLimit gets cross/isolated margin risk limit, default model is cross margin
+func (ku *Kucoin) GetMarginRiskLimit(ctx context.Context, marginModel string) ([]MarginRiskLimit, error) {
+	resp := struct {
+		Data []MarginRiskLimit `json:"data"`
+		Error
+	}{}
+	params := url.Values{}
+	if marginModel != "" {
+		params.Set("marginModel", marginModel)
+	}
+	return resp.Data, ku.SendAuthHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, common.EncodeURLValues(kucoinGetMarginRiskLimit, params), nil, &resp)
+}
+
+// PostBorrowOrder used to post borrow order
+func (ku *Kucoin) PostBorrowOrder(ctx context.Context, currency, orderType, term string, size, maxRate float64) (*PostBorrowOrderResp, error) {
+	resp := struct {
+		Data PostBorrowOrderResp `json:"data"`
+		Error
+	}{}
+	params := make(map[string]interface{})
+	if currency == "" {
+		return nil, errors.New("currency can't be empty")
+	}
```

Comment:
These checks can be shifted to the top of the function and declarations below that. Please change across this PR.